src/Test.UnitTests.Sarif/Core/ResultTests.cs

```diff
@@ -43,6 +48,95 @@ public void Result_Level_ResetsKindValue()
             result.Kind.Should().Be(ResultKind.Fail);
         }
 
+        [Fact]
+        public void Result_Level_WriteCorrectValue_GeneralCases()
```

Comment:
>Result_Level_WriteCorrectValue_GeneralCases [](http://example.com/codeflow?start=20&length=43)

The same new tests are also added in a separated branch based on a main to test how it looks, not all tests can pass,
link:
[https://github.com/microsoft/sarif-sdk/pull/2409](https://github.com/microsoft/sarif-sdk/pull/2409)